apps/src/templates/manageStudents/SharingControlActionsHeaderCell.jsx

```diff
@@ -1,9 +1,25 @@
-import React, {Component} from 'react';
+import React, {Component, PropTypes} from 'react';
 import QuickActionsCell, {QuickActionsCellType} from "../tables/QuickActionsCell";
 import PopUpMenu, {MenuBreak} from "@cdo/apps/lib/ui/PopUpMenu";
 import i18n from '@cdo/locale';
+import {updateAllShareSetting, editAll} from './manageStudentsRedux';
+import {connect} from 'react-redux';
 
-export default class SharingControlActionsHeaderCell extends Component {
+class SharingControlActionsHeaderCell extends Component {
+  static propTypes = {
+    updateAllShareSetting: PropTypes.func,
+    editAll: PropTypes.func,
+  };
+
+  onEnableAll = () => {
+    this.props.editAll();
```

Comment:
While this makes the code work, the reason it's better to use a redux thunk over creating both actions from the component is that `updateAllShareSetting` is not a valid action to do alone (since it'll error), so you only want to expose it externally when combined with an action of type`EDIT_ALL`.

You can always still test an action with type `UPDATE_ALL_SHARE_SETTING` on it's own, or in combination with `EDIT_ALL` action.

Redux thunk explanation: https://daveceddia.com/what-is-a-thunk/

Reply:
oooooh.  Is it still okay to do something similar here: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/manageStudents/ControlProjectSharingDialog.jsx#L34 because the `editAll` and `toggleShareColumn` can be successful independently?

Reply:
exactly.